packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionStatusInfo.tsx

```diff
@@ -4,101 +4,154 @@
 */
 
 import { useAccount } from 'wagmi'
-import { useMemo } from 'react'
-import {
-  CheckCircleIcon,
-  InformationCircleIcon
-} from '@heroicons/react/24/outline'
+import { useCallback, useMemo } from 'react'
+import { ExclamationTriangleIcon } from '@heroicons/react/24/outline'
 import { twMerge } from 'tailwind-merge'
+import Image from 'next/image'
+import ArrowsIcon from '@/images/arrows.svg'
 
-import {
-  isDepositReadyToRedeem,
-  isWithdrawalReadyToClaim
-} from '../../state/app/utils'
-import { trackEvent } from '../../util/AnalyticsUtils'
+import { isDepositReadyToRedeem } from '../../state/app/utils'
 import { useAppContextActions } from '../App/AppContext'
-import { ExternalLink } from '../common/ExternalLink'
 import { useTransactionHistory } from '../../hooks/useTransactionHistory'
+import { Button } from '../common/Button'
+import { isTxClaimable, isTxPending } from './helpers'
+import { Transition } from '../common/Transition'
 
 export const TransactionStatusInfo = () => {
   const { address } = useAccount()
   const { openTransactionHistoryPanel } = useAppContextActions()
   const { transactions } = useTransactionHistory(address)
 
-  const { numWithdrawalsReadyToClaim, numRetryablesToRedeem } = useMemo(() => {
+  const {
+    numClaimableTransactions,
+    numRetryablesToRedeem,
+    numPendingTransactions
+  } = useMemo(() => {
     return transactions.reduce(
       (acc, tx) => {
         // standard bridge withdrawal
-        if (isWithdrawalReadyToClaim(tx)) {
-          acc.numWithdrawalsReadyToClaim += 1
+        if (isTxClaimable(tx)) {
+          acc.numClaimableTransactions += 1
         }
         // failed retryable
         if (isDepositReadyToRedeem(tx)) {
           acc.numRetryablesToRedeem += 1
         }
+        // all pending
+        if (isTxPending(tx)) {
+          acc.numPendingTransactions += 1
+        }
         return acc
       },
-      { numWithdrawalsReadyToClaim: 0, numRetryablesToRedeem: 0 }
+      {
+        numClaimableTransactions: 0,
+        numRetryablesToRedeem: 0,
+        numPendingTransactions: 0
+      }
     )
   }, [transactions])
 
-  // don't show this banner if user doesn't have anything to claim or redeem
-  if (numWithdrawalsReadyToClaim === 0 && numRetryablesToRedeem === 0)
+  const shouldShow = useMemo(() => {
+    return (
+      numClaimableTransactions > 0 ||
+      numRetryablesToRedeem > 0 ||
+      numPendingTransactions > 0
+    )
+  }, [numClaimableTransactions, numRetryablesToRedeem, numPendingTransactions])
+
+  const Content = useCallback(() => {
+    const numClaimableTransactionsString = `claim ${numClaimableTransactions} ${
+      numClaimableTransactions === 1 ? 'transaction' : 'transactions'
```

Comment:
can we move it to a nice looking util function and call it to determine if we need the `s`?
this is annoying